rbi/annotations/graphql.rbi

```diff
@@ -38,7 +38,7 @@ module GraphQL::Schema::Member::HasFields
     params(
       args: T.untyped,
       kwargs: T.untyped,
-      block: T.nilable(T.proc.bind(GraphQL::Schema::Field).void)
+      block: T.nilable(T.proc.params(field: GraphQL::Schema::Field).bind(GraphQL::Schema::Field).void)
```

Comment:
Here's a dumbed-down example that shows that both calling conventions are supported:

[Sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Aextend%20T%3A%3ASig%0A%0Aclass%20Field%0A%20%20def%20argument%3B%20end%0Aend%0A%0Asig%20do%0A%20%20params%28%0A%20%20%20%20block%3A%20T.nilable%28%0A%20%20%20%20%20%20T.proc%0A%20%20%20%20%20%20%20%20.params%28field%3A%20Field%29%0A%20%20%20%20%20%20%20%20.bind%28Field%29%0A%20%20%20%20%20%20%20%20.void%0A%20%20%20%20%29%0A%20%20%29.void%0Aend%0Adef%20field%28%26block%29%0A%20%20block%26.call%28Field.new%29%0Aend%0A%0Afield%20do%0A%20%20argument%0Aend%0A%0Afield%20do%20%7Cfield%7C%0A%20%20field.argument%0A%20%20argument%0Aend)

```ruby
# typed: true
extend T::Sig

class Field
  def argument; end
end

sig do
  params(
    block: T.nilable(
      T.proc
        .params(field: Field)
        .bind(Field)
        .void
    )
  ).void
end
def field(&block)
  block&.call(Field.new)
end

field do
  argument
end

field do |field|
  field.argument
  argument
end
```